python/vowpalwabbit/pyvw.py

```diff
@@ -1398,386 +1432,413 @@ def get_label(self, label_class=simple_label):
-    attr_values = list(vars(obj).values())
-    out = [x for x in attr_values if isinstance(x, Col)]
-    return out
+    def initialize_col_type(self, instance, colname):
+        """Initialize the attribute as a _Col type
 
+        Parameters
+        ----------
+        instance: object
+            The managed instance.
+        colname: str/int/float
+            The colname used for the _Col instance.
+        """
 
-def _check_type(obj, expected_type):
-    """Check if the type of an object is valid.
+        instance.__dict__["columns"].add(colname)
+        instance.__dict__[self.attribute_name] = _Col(
+            colname=colname,
+            expected_type=self.expected_type,
+            min_value=self.min_value,
+        )
 
-    Parameters
-    ----------
+    def generate_error(self, instance, valid_type, valid_value):
+        """Generate message error for the attribute
 
-    obj : object
-        The object to check.
-    expected_type : type or tuple of types
-        The type(s) to check against.
+        Parameters
+        ----------
+        instance: object
+            The managed class.
+        valid_type: bool
+            True if the type if valid, False otherwise.
+        valid_value: bool
+            True if the value range if valid, False otherwise.
 
-    Raises
-    ------
+        Raises
+        ------
+        TypeError
+            If the type is not valid.
+        ValueRange
+            If the value range is not valid.
+        """
+        class_name = type(instance).__name__
 
-    TypeError
-        If the argument is not of a valid type.
-    """
-    expected_type_str = str([x.__name__ for x in expected_type])
-    if obj is not None:
-        if not isinstance(obj, expected_type):
+        if not valid_type:
             raise TypeError(
-                "Parameter {} should be of type(s) {}".format(
-                    obj, expected_type_str[1:-1]
+                "In '{class_name}', when {from_df_false}, argument '{attribute_name}' should be either of the following type(s): {type_name_msg}.".format(
+                    class_name=class_name,
+                    from_df_false=self.attribute_name + "_from_df=False",
+                    attribute_name=self.attribute_name,
+                    type_name_msg=str(
+                        [x.__name__ for x in self.expected_type]
+                    )[1:-1],
+                )
+            )
+
+        if not valid_value:
+            raise ValueError(
+                "In '{class_name}', argument '{attribute_name}' must be >= {min_value}.".format(
+                    class_name=class_name,
+                    attribute_name=self.attribute_name,
+                    min_value=self.min_value,
                 )
             )
 
 
-class SimpleLabel:
-    """The SimpleLabel class is used to build a simple label for the
-    constructor of DFtoVW.
-    """
+class SimpleLabel(object):
+    """The simple label type for the constructor of DFtoVW."""
```

Comment:
Not related to this PR, but this naming convention might be confusing since this is only helpful for DFtoVW. What do you think @jackgerrits

Reply:
Yep, I saw too that in pyvw there are classes like `simple_label`. In my opinion the following 2 things could be done : 
- prepending all the classes used in `DFtoVW` with `DF` 
- since `simple_label` and friends does not respect [python's CapsWords convention for class names](https://www.python.org/dev/peps/pep-0008/#class-names) it could be renamed with the appropriate CapsWords. Also, it won't clash with the classes used in DFtoVW since we preppend them with "DF"